discord/automoderation.go

```diff
@@ -0,0 +1,90 @@
+package discord
+
+import "time"
+
+type AutoModerationEventTypes uint32
+
+const (
+	AutoModerationMessageSend AutoModerationEventTypes = 1 + iota
+	AutoModerationMemberUpdate
+)
+
+type AutoModerationTriggerTypes uint32
+
+const (
+	AutoModerationKeyword AutoModerationTriggerTypes = 1 + iota
+	AutoModerationSpam
+	AutoModerationKeywordPreset
+	AutoModerationMentionSpam
+	AutoModerationMemberProfile
+)
+
+type AutoModerationKeywordPresetTypes uint32
+
+const (
+	AutoModeratorProfanity = 1 + iota
+	AutoModeratorSexualContent
+	AutoModeratorSlurs
+)
+
+type AutoModerationRuleID Snowflake
```

Comment:
For the snowflake types, you'd want to add it into `snowflake.go`'s `go:generate` line the re-generate the code via `go generate ./...`.

Reply:
Done in [e1b274b](https://github.com/diamondburned/arikawa/pull/444/commits/e1b274bc74e9af7d3ee1fa6bef8cfc2a73f644ce)